### ISSUE TITLE
CoreWebView2Frame.ExecuteScriptAsync hangs forever if the frame's destroyed while script is executing

### ISSUE THREAD

**Issue by q71114** [MicrosoftEdge/WebView2Feedback#3124]
**Description**
`CoreWebView2Frame.ExecuteScriptAsync` hangs if the frame is destroyed while script is executing. _Forever_ is in the title but obviously I haven't left the app running forever, however after 1.5 hour of testing it is still hanging.

**Version**
SDK: 1.0.1549-prerelease
Runtime: Evergreen (108.0.1462.76)
Framework: WPF
OS: Win11 22H2 (OS Build 22621.521)

**Repro Steps**
1. Set up a script when a frame is created and loading to post web messages periodically.
2. Handle `CoreWebView2Frame.WebMessageReceived` and call `CoreWebView2Frame.ExecuteScriptAsync` in it
3. Or alternatively just clone [this branch](https://github.com/q71114/WebView2Samples/tree/bug/frame-executescriptasync-hanging) I created from the WebView2Samples.
4. Navigate to any website that has a few iframes. I tested with https://reddit.com
5. Open another website that also has a few iframes. I tested by clicking on any link on reddit.
6. Click Back / Forward a few times to alternate between the two websites while checking the Output window

**Expected**: All scripts execution should complete
**Actual**: When a frame is destroyed right after script is being executed the script execution neither completes nor throws any exception even after 1.5 hours (possibly indefinitely).

**Additional context**
When this occurs you should see something like this in the Output window when using the [test project](https://github.com/q71114/WebView2Samples/tree/bug/frame-executescriptasync-hanging) I provided
```
Before execute script in frame ''
Destroyed: frame ''
...
Currently waiting for 1 script(s) to finish
```

[AB#43041545](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/43041545)

**Comment by david-risney** [comment]
Thanks for the detailed bug report and including the code - that's very helpful! I expect it is forever like you speculate. 

Similar sorts of problems I would expect also exist for async APIs on CoreWebView2 although having this happen if a frame happens to navigate during an async call should make this instance of the issue much more prevalent. We will look into this, thanks!

**Comment by vbryh-msft** [comment]
This is fixed in 116.0.1923.0+
